internal/manifests/collector/parser/receiver/generic_receivers.go

```diff
@@ -0,0 +1,169 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receiver
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/parser"
+)
+
+const (
+	unsetPort = 0
+)
+
+var (
+	grpc             = "grpc"
+	http             = "http"
+	scraperReceivers = map[string]struct{}{
+		"prometheus":        {},
+		"kubeletstats":      {},
+		"sshcheck":          {},
+		"cloudfoundry":      {},
+		"vcenter":           {},
+		"oracledb":          {},
+		"snmp":              {},
+		"googlecloudpubsub": {},
+		"chrony":            {},
+		"jmx":               {},
+		"podman_stats":      {},
+		"pulsar":            {},
+		"docker_stats":      {},
+		"aerospike":         {},
+		"zookeeper":         {},
+		"prometheus_simple": {},
+		"saphana":           {},
+		"riak":              {},
+		"redis":             {},
+		"rabbitmq":          {},
+		"purefb":            {},
+		"postgresql":        {},
+		"nsxt":              {},
+		"nginx":             {},
+		"mysql":             {},
+		"memcached":         {},
+		"httpcheck":         {},
+		"haproxy":           {},
+		"flinkmetrics":      {},
+		"couchdb":           {},
+	}
+	genericReceivers = map[string][]parser.SinglePortOption{
```

Comment:
Pretty much all of our receivers can be done without any customization beyond what's here